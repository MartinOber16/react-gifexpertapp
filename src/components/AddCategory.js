import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );

        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if( inputValue.trim().length > 2) {
            //props.setCategories( cats => [ ...cats, inputValue] );
            setCategories( cats => [ inputValue, ...cats] ); // Para ordenar en forma inversa
            setInputValue('');
            console.log('Submit hecho');
        }

    }

    return (
        <form onSubmit={ handleSubmit } >
            <p>{ inputValue }</p>
           <input 
                onChange={ handleInputChange }
                type="text"
                value={ inputValue }
           />
        </ form>
    )
}

// La funcion es obligatoria
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;