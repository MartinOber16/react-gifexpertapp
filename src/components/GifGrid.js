// https://developers.giphy.com/docs/api/endpoint#search

//import React, { useEffect, useState } from 'react'
import React from 'react'
import PropTypes from 'prop-types'; // impt
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category }) => {

    //const [images, setImages] = useState([]);
    const { data: images, loading } = useFetchGifs( category ); // custom hook

    // Ejecuta solo cuando el componente es renderizado por primera vez
    // useEffect( () => {
    //     getGifs( category )
    //         //.then( imgs => setImages( imgs ) );
    //         .then( setImages );
    // }, [ category ]);

    // const getGifs = async () => {

    //     const apiURL = 'api.giphy.com/v1/gifs/search';
    //     const apiKey = '';
    //     const apiLimit = 10;
    //     const apiQuery = encodeURI( category );

    //     const url = `https://${apiURL}?q=${apiQuery}&limit=${apiLimit}&api_key=${apiKey}`;
    //     const resp = await fetch( url );
    //     const { data } = await resp.json();
    //     const gifs = data.map( img => {
    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url // el ? es por si no viene nada
    //         }
    //     })

    //     //console.log( gifs );
    //     setImages( gifs );

    // }

    return (
        <>
            <h3 className="animate__animated animate__fadeIn" > { category } </h3>

            { loading && <p className="animate__animated animate__flash" >Loading</p> }

            <div className="card-grid" >
                {/* <ol> */}
                    { 
                        // images.map( imagen =>  <li key={imagen.id} >{imagen.title} </li> )
                        images.map( 
                            //( {id, title } ) => {
                            ( img ) => {
                                //return <li key={id} > {title} </li>
                            return <GifGridItem 
                                    key={img.id} 
                                    // img={img}
                                    { ...img } // envio cada una de las propiedades de img como independientes
                                />
                            } 
                        )
                    }
                {/* </ol> */}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
