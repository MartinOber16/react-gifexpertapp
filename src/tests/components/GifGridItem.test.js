import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem.js', () => {

    const id = "123";
    const title = "Titulo";
    const url = "http://prueba/123";
    const wrapper = shallow( <GifGridItem id={ id } title={ title } url={ url } /> );

    test('Debe mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    }); 

    test('Debe tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    }); 

    test('Debe tener la imagen igual al url y alt de los prop', () => {
        
        const img = wrapper.find('img');
        //console.log( img.props() );
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );
        
    }); 

    test('Debe tener la clase animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop( 'className');

        expect( className.includes('animate__fadeIn') ).toBe( true );
    })
    
    
});
