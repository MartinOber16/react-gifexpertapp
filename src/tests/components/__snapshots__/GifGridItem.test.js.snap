// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="http://prueba/123"
  />
  <p>
     
    Titulo
     
  </p>
</div>
`;
